src/redis-cli.c

```diff
@@ -3651,7 +3671,11 @@ static int clusterManagerFixSlotsCoverage(char *all_slots) {
                 clusterManagerLogInfo(">>> Covering slot %s moving keys "
                                       "to %s:%d\n", slot,
                                       target->ip, target->port);
+                /* Ensure the slot is not already assigned. */
                 redisReply *r = CLUSTER_MANAGER_COMMAND(target,
+                    "CLUSTER DELSLOTS %s", slot);
```

Comment:
Same as in "single" case: doesn't it delete data?

Reply:
DELSLOTS just unassigns the slot in the node configuration. It's used before ADDSLOTS (in case the node must become the "owner") or before SETSLOT (when the node must update its own configuration after the "owner" changed) in order to avoid reply errors related to the fact the the slot is already assigned from the node's point of view.

No key will be deleted, it's just used to reassign the slot in the node's configuration.